pkg/ruler/compat.go

```diff
@@ -129,12 +187,30 @@ type RulesManager interface {
 // ManagerFactory is a function that creates new RulesManager for given user and notifier.Manager.
 type ManagerFactory func(ctx context.Context, userID string, notifier *notifier.Manager, logger log.Logger, reg prometheus.Registerer) RulesManager
 
-func DefaultTenantManagerFactory(cfg Config, p Pusher, q storage.Queryable, engine *promql.Engine, overrides RulesLimits) ManagerFactory {
+func DefaultTenantManagerFactory(cfg Config, p Pusher, q storage.Queryable, engine *promql.Engine, overrides RulesLimits, globalReg prometheus.Registerer) ManagerFactory {
+	totalWrites := promauto.With(globalReg).NewCounter(prometheus.CounterOpts{
+		Name: "cortex_ruler_write_requests_total",
+		Help: "Number of write requests to ingesters.",
+	})
+	failedWrites := promauto.With(globalReg).NewCounter(prometheus.CounterOpts{
+		Name: "cortex_ruler_write_requests_failed_total",
+		Help: "Number of failed write requests to ingesters.",
+	})
+
+	totalQueries := promauto.With(globalReg).NewCounter(prometheus.CounterOpts{
+		Name: "cortex_ruler_queries_total",
+		Help: "Number of queries executed by ruler.",
+	})
+	failedQueries := promauto.With(globalReg).NewCounter(prometheus.CounterOpts{
+		Name: "cortex_ruler_queries_failed_total",
+		Help: "Number of failed queries by ruler.",
+	})
+
```

Comment:
**question:** why use  the name globalReg here? Is this to let let the caller know not to use a per-tenant reg?

```suggestion
func DefaultTenantManagerFactory(cfg Config, p Pusher, q storage.Queryable, engine *promql.Engine, overrides RulesLimits, reg prometheus.Registerer) ManagerFactory {
	totalWrites := promauto.With(reg).NewCounter(prometheus.CounterOpts{
		Name: "cortex_ruler_write_requests_total",
		Help: "Number of write requests to ingesters.",
	})
	failedWrites := promauto.With(reg).NewCounter(prometheus.CounterOpts{
		Name: "cortex_ruler_write_requests_failed_total",
		Help: "Number of failed write requests to ingesters.",
	})

	totalQueries := promauto.With(reg).NewCounter(prometheus.CounterOpts{
		Name: "cortex_ruler_queries_total",
		Help: "Number of queries executed by ruler.",
	})
	failedQueries := promauto.With(reg).NewCounter(prometheus.CounterOpts{
		Name: "cortex_ruler_queries_failed_total",
		Help: "Number of failed queries by ruler.",
	})

```

Reply:
> **question:** why use the name globalReg here? Is this to let let the caller know not to use a per-tenant reg?

Yes. I wanted to make it explicit that registry here is not the same as registry used by function in the return statement (which is per-tenant). But I guess it's clear enough already.